src/components/twilio-voice-dialer/public/twilio-voice-dialer.js

```diff
@@ -1 +1,51 @@
-// TODO: Implement Web Component
+const render = (x) => `
+  <div class="container">
+    <p id="status">Status: pending</p>
+    <input type="text" placeholder="recipient" id="recipient" value=${x.recipient}>
+  </div>
+`;
+
+class TwilioVoiceDialer extends HTMLElement {
+  static get observedAttributes() {
+    return ['recipient'];
+  }
+
+  #device;
+  #token;
+
+  constructor() {
+    super();
+    this.attachShadow({ mode: 'open' });
+  }
+  get recipient() {
+    return this.getAttribute('recipient');
+  }
+
+  connectedCallback() {}
+
+  disconnectedCallback() {
+    this.#device.destroy();
+  }
+
+  attributeChangedCallback() {
+    this.shadowRoot.innerHTML = render(this);
```

Comment:
I think it's better if we create a private render function that sets the innerHTML. Then inside attributeChangedCallback, we can call `this.render()`;